File: web/src/services/reportService.ts
// web/src/services/reportService.ts

// UNUSED: Only called by generateReportV1 which is also dead
export function _buildHeader(title: string): string {
  return `=== ${title.toUpperCase()} ===`;
}

// UNUSED: only called by generateReportV1 which is also dead
export function _buildFooter(title: string): string {
  return `--- ${title} ---`;
}

// UNUSED: never called from anywhere in the project
export function generateReportV1(title: string, body: string): string {
  const header = _buildHeader(title);
  const footer = _buildFooter(title);
  return `${header}\n${body}\n${footer}`;
}

// UNUSED: called inside an `if (false)` branch that never executes
export function _searchV2(query: string): string[] {
  return [`v2-result-for-${query}`];
}

export function search(query: string): string[] {
  const enableV2 = false;
  if (enableV2) {
    return _searchV2(query);
  }
  return [`basic-result-for-${query}`];
}

File: app/services/report_service.py
def _build_header(title: str) -> str:
    ## UNUSED: Only called by generate_report_v1 which is also dead
    return f"=== {title.upper()} ==="


def _build_footer(title: str) -> str:
    ## UNUSED: only called by generate_report_v1 which is also dead
    return f"--- {title} ---"


def generate_report_v1(title: str, body: str) -> str:
    ## UNUSED: never called from anywhere in the project
    header = _build_header(title)
    footer = _build_footer(title)
    return f"{header}\n{body}\n{footer}"


def _search_v2(query: str) -> list:
    ## UNUSED: called inside an `if False:` branch that never executes
    return [f"v2-result-for-{query}"]


def search(query: str) -> list:
    if False:  # was: if ENABLE_V2_SEARCH
        return _search_v2(query)
    return [f"basic-result-for-{query}"]
